### ISSUE TITLE
[TEST-FAILURE] CacheBasicClientTest.testCompletionEvent

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#6622]
```
java.lang.AssertionError: expected:<2> but was:<4>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.cache.CacheBasicAbstractTest$12.run(CacheBasicAbstractTest.java:554)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:825)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:839)
	at com.hazelcast.cache.CacheBasicAbstractTest.testCompletionEvent(CacheBasicAbstractTest.java:551)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OracleJDK1.7/com.hazelcast$hazelcast-client-legacy/287/testReport/junit/com.hazelcast.client.cache/CacheBasicClientTest/testCompletionEvent/

**Comment by hasancelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-IbmJDK1.6/com.hazelcast$hazelcast-client-legacy/736/testReport/junit/com.hazelcast.client.cache/CacheBasicClientTest/testCompletionEvent/

**Comment by sancar** [comment]
fixed by #6728
